two small-ish fixes

FrankyBoy opened the pull request:
for #1 and #3 (tbd)

FrankyBoy committed:
only use the https agent if the url is https

FrankyBoy committed:
extend options rather than overwriting them

FrankyBoy committed:
workaround for #4 ... don't use "options.domain" because request.js already uses that for something (which fails)

FrankyBoy committed:
fixes #5 ... don't assume everything is json

colynb commented:
Thanks for this @FrankyBoy - A large team is currently using this for a large build process that uses this for connecting to an older api (which is why I built this) so I'll need to do some testing for a bit before I can get this merged. But thanks for your contribution!

FrankyBoy commented:
Jeah, totally. I am just currently writing node bindings for some SOAP-services that use ntlm (the joy ... both to soap and ntlm) so I'm just patching in stuff as I go along to suit my needs while trying not to break anything major ;)

FrankyBoy committed:
as the fixes are still not in upstream, gonna publish this on my own now.

FrankyBoy committed:
Create LICENSE.txt

brettp committed:
Fixed bug that prevented type1 message auth

The headers property must be set before being extended with the type1 auth.

FrankyBoy committed:
Merge pull request #1 from brettp/brettp-extend-headers

Fixed bug that prevented type1 message auth

FrankyBoy committed:
push version to 0.1.1, add changelog

garthkerr committed:
Upgrade agentkeepalive dependency for node 0.12

garthkerr committed:
Upgrade package version

FrankyBoy committed:
Merge pull request #2 from garthkerr/master

Upgrade agentkeepalive dependency for node 0.12

Commit:
allow requests to non-NTLM resources + tests

Commit:
adding .travis.yml

Commit:
fixing test case without ntlm credentials

FrankyBoy committed:
Merge pull request #4 from AndreasPizsa/master

Allow non-NTLM resources

adam-nielsen committed:
Make request.pipe() accessible for streaming

FrankyBoy committed:
Merge pull request #5 from adam-nielsen/master

Make request.pipe() accessible for streaming

Commit:
Push version to 0.1.4 before release

ygalbel committed:
Add patch verb

Hi, I had to use PATCH request for some reasons.

Add exports from request.

Works fine.
Can you merge it?

FrankyBoy committed:
Merge pull request #6 from ygalbel/master

Add patch verb

Commit:
Push version to 0.1.5 before release, also align indentation in the exports section.
